front/lib/api/assistant/configuration.ts

```diff
@@ -1149,6 +1155,42 @@ export async function createAgentActionConfiguration(
         forceUseAtIteration: action.forceUseAtIteration,
       };
     });
+  } else if (action.type === "process_configuration") {
+    return frontSequelize.transaction(async (t) => {
+      const processConfig = await AgentProcessConfiguration.create(
+        {
+          sId: generateModelSId(),
+          relativeTimeFrame: isTimeFrame(action.relativeTimeFrame)
+            ? "custom"
+            : action.relativeTimeFrame,
+          relativeTimeFrameDuration: isTimeFrame(action.relativeTimeFrame)
+            ? action.relativeTimeFrame.duration
+            : null,
+          relativeTimeFrameUnit: isTimeFrame(action.relativeTimeFrame)
+            ? action.relativeTimeFrame.unit
+            : null,
+          agentConfigurationId: agentConfiguration.id,
+          schema: action.schema,
+          forceUseAtIteration: action.forceUseAtIteration,
+        },
+        { transaction: t }
+      );
+      await _createAgentDataSourcesConfigData(
+        t,
+        action.dataSources,
+        processConfig.id
+      );
+
+      return {
+        id: processConfig.id,
+        sId: processConfig.sId,
+        type: "process_configuration",
+        relativeTimeFrame: action.relativeTimeFrame,
+        schema: action.schema,
+        dataSources: action.dataSources,
+        forceUseAtIteration: action.forceUseAtIteration,
+      };
+    });
   } else {
     throw new Error("Cannot create AgentActionConfiguration: unknow type");
```

Comment:
can we assert never here or `type` is just a string ?

Reply:
sure